core/dbt/tests/util.py

```diff
@@ -93,7 +94,7 @@ def run_dbt(
         args.extend(["--project-dir", project_dir])
     if profiles_dir and "--profiles-dir" not in args:
         args.extend(["--profiles-dir", profiles_dir])
-    dbt = dbtRunner()
+    dbt = dbtRunner(callbacks=callbacks)
```

Comment:
It looks like this pattern is used in many places, let's refactor them all at once?

For now, this should be fine?

```
@pytest.fixture(scope="function")
def runner(catcher: EventCatcher) -> dbtRunner:
    return dbtRunner(callbacks=[catcher.catch])
```

Reply:
That's a good idea. However, we wouldn't be able to use that in this util. I think that fixture would likely make most sense `tests/` not `core/dbt/tests/`. This utility function (`run_dbt`) wouldn't have access to it. That doesn't mean we shouldn't do that work, but it's OOO for this PR and should be its own segment of work.